x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/schema/TransformConfigTests.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.transform.transforms.schema;
+
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.ToXContent.Params;
+import org.elasticsearch.test.AbstractSchemaValidationTestCase;
+import org.elasticsearch.xpack.core.transform.TransformField;
+import org.elasticsearch.xpack.core.transform.transforms.TransformConfig;
+
+import java.util.Collections;
+
+import static org.elasticsearch.xpack.core.transform.transforms.TransformConfigTests.randomTransformConfig;
+
+public class TransformConfigTests extends AbstractSchemaValidationTestCase<TransformConfig> {
+
+    protected static Params TO_XCONTENT_PARAMS = new ToXContent.MapParams(
+        Collections.singletonMap(TransformField.EXCLUDE_GENERATED, "true")
```

Comment:
I wonder if this line reveals a major problem for the usefulness of the schemas to the clients team.

The clients team will need to create clients that can parse and provider accessors for the generated fields as well as the ones used to initially create the transform config.  If the schema doesn't include them then any generated clients will be incomplete.

It seems to me that the schema needs to include all fields that might ever be included with the most expansive options.  In this case that will mean `create_time` and `version`.  But then I wonder if that plays havoc with stating what is required and what is optional in the schema?  Certain fields may be required if and only if a certain option was provided to the endpoint that returned the document.

Reply:
see https://github.com/elastic/elasticsearch/pull/69902/files#diff-ac85f75f4c81bedfc8e3bc397dd47267d6e2f83cc8f808c2b3ff61b4c627c15bR25, it specs the transform `_stats` response by re-using the spec for the stats object.

However, some ES API's behave different, depending on parameters. We might find _corner cases_ where the schema is not able to correctly describe it.